edx_name_affirmation/signals.py

```diff
@@ -0,0 +1,179 @@
+# pylint: disable=logging-format-interpolation
+
+""""
+Signal handlers for IDV and proctoring service signals
+"""
+
+import logging
+
+from django.contrib.auth import get_user_model
+
+from edx_name_affirmation.models import VerifiedName
+from edx_name_affirmation.statuses import VerifiedNameStatus
+from edx_name_affirmation.toggles import is_verified_name_enabled
+
+try:
+    from lms.djangoapps.verify_student.signals import idv_update_signal
+except ImportError:
+    idv_update_signal = None
+
+try:
+    from edx_proctoring.utils import exam_attempt_status_signal
+except ImportError:
+    exam_attempt_status_signal = None
+
+User = get_user_model()
+
+log = logging.getLogger(__name__)
+
+
+def update_verified_name_for_idv_attempt(attempt_id, user_id, status, full_name, profile_name, **kwargs):
+    """
+    Receiver for IDV attempt updates
+
+    Args:
+        attempt_id(int): ID associated with the IDV attempt
+        user_id(int): ID associated with the IDV attempt's user
+        status(str): status in IDV language for the IDV attempt
+        full_name(str): name to be used as verified name
+        profile_name(str): user's current profile name
+    """
+    translated_status = VerifiedNameStatus.translate_idv_status(status)
+    log_str = ''
+
+    verified_name = VerifiedName.objects.filter(
+        user__id=user_id,
+        verification_attempt_id=attempt_id
+    ).order_by('-created').first()
+    if verified_name:
```

Comment:
I feel like the way you organized the `if` and `else` block is slightly confusing. 
What if we first get an array of verified_name data set which matches the user_id and the actual full name? Then for each such record, if the attempt_id on the row is not specified, assign it to the passed in attempt_id, and update its status if the translated status is available? 
Then you can handle the case when array from the matching query is empty (create). That makes the code easier to read?